File: frontend/src/types/project.ts
// ============================================================
// 项目模块类型（M06 深度开发：里程碑 + 统计 + 详情聚合）
// ============================================================

export type ProjectStatus = 'active' | 'archived' | 'completed'

export interface Project {
  id: string
  user_id: string
  name: string
  description: string | null
  color: string
  status: ProjectStatus
  due_date: string | null
  sort_order: number
  task_count: number
  completed_count: number
  progress: number
  created_at: string
  updated_at: string
}

export interface CreateProjectRequest {
  name: string
  description?: string
  color?: string
  status?: ProjectStatus
  due_date?: string
}

export interface UpdateProjectRequest {
  name?: string
  description?: string
  color?: string
  status?: ProjectStatus
  due_date?: string
  sort_order?: number
}

// ---------- 里程碑（M06 F03） ----------

export interface Milestone {
  id: string
  project_id: string
  name: string
  description: string | null
  target_date: string | null
  completed: boolean
  completed_at: string | null
  task_ids: string[]
  sort_order: number
  created_at: string
  updated_at: string
}

export interface MilestoneCreateRequest {
  name: string
  description?: string
  target_date?: string
  task_ids?: string[]
  sort_order?: number
}

export interface MilestoneUpdateRequest {
  name?: string
  description?: string
  target_date?: string
  completed?: boolean
  task_ids?: string[]
  sort_order?: number
}

// ---------- 项目统计（M06 F02） ----------

export interface ProjectStats {
  task_count: number
  completed_count: number
  progress: number
  status_distribution: Record<string, number>
  weekly_trend: Array<{ date: string; completed: number }>
  overdue_count: number
  milestone_count: number
  milestone_completed: number
  health: 'good' | 'warning' | 'risk'
}

// ---------- 项目详情聚合（M06 F01） ----------

export interface ProjectDetailTask {
  id: string
  title: string
  status: string
  priority: string
  due_date: string | null
  is_focus: boolean
  completed_at: string | null
  created_at: string | null
}

export interface ProjectDetailDocument {
  id: string
  title: string
  updated_at: string | null
  tags: string[]
  word_count: number
}

export interface ProjectDetailConversation {
  id: string
  title: string
  updated_at: string | null
}

export interface ProjectDetailReview {
  id: string
  type: string
  review_date: string
  mood: number | null
  summary: string
}

export interface ProjectDetail extends Project {
  tasks: ProjectDetailTask[]
  documents: ProjectDetailDocument[]
  conversations: ProjectDetailConversation[]
  reviews: ProjectDetailReview[]
  milestones: Milestone[]
}

File: backend/app/schemas/project.py
# ============================================================
# Project Schema
# ============================================================
from __future__ import annotations

from datetime import date
from typing import Optional

from pydantic import BaseModel, Field


class ProjectBase(BaseModel):
    name: str = Field(..., max_length=200)
    description: Optional[str] = None
    color: str = Field(default="#7c5cff", max_length=20)
    status: str = Field(default="active", max_length=20)
    due_date: Optional[date] = None


class ProjectCreate(ProjectBase):
    pass


class ProjectUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    color: Optional[str] = Field(None, max_length=20)
    status: Optional[str] = Field(None, max_length=20)
    due_date: Optional[date] = None
    sort_order: Optional[int] = None


class ProjectOut(ProjectBase):
    id: str
    user_id: str
    sort_order: int
    task_count: int = 0
    completed_count: int = 0
    progress: int = 0
    created_at: str
    updated_at: str

    class Config:
        from_attributes = True


# ---------- 里程碑（M06 F03） ----------

class MilestoneCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=200)
    description: Optional[str] = None
    target_date: Optional[date] = None
    task_ids: list[str] = []
    sort_order: int = 0


class MilestoneUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=200)
    description: Optional[str] = None
    target_date: Optional[date] = None
    completed: Optional[bool] = None
    task_ids: Optional[list[str]] = None
    sort_order: Optional[int] = None


class MilestoneOut(BaseModel):
    id: str
    project_id: str
    name: str
    description: Optional[str] = None
    target_date: Optional[date] = None
    completed: bool = False
    completed_at: Optional[str] = None
    task_ids: list[str] = []
    sort_order: int = 0
    created_at: str
    updated_at: str


# ---------- 项目统计（M06 F02） ----------

class ProjectStatsOut(BaseModel):
    """项目进度统计：完成率 + 状态分布 + 近7天趋势 + 逾期 + 健康度。"""

    task_count: int = 0
    completed_count: int = 0
    progress: int = 0
    status_distribution: dict[str, int] = {}
    weekly_trend: list[dict] = []  # [{date, completed}]
    overdue_count: int = 0
    milestone_count: int = 0
    milestone_completed: int = 0
    health: str = "good"  # good / warning / risk
